src/scraper.py

```diff
@@ -0,0 +1,163 @@
+import sys
+import requests
+import re
+from bs4 import BeautifulSoup
+import io
+
+INSPECTION_DOMAIN = 'http://info.kingcounty.gov'
+INSPECTION_PATH = '/health/ehs/foodsafety/inspections/Results.aspx'
+INSPECTION_PARAMS = {
+    'Output': 'W',
+    'Business_Name': '',
+    'Business_Address': '',
+    'Longitude': '',
+    'Latitude': '',
+    'City': '',
+    'Zip_Code': '',
+    'Inspection_Type': 'All',
+    'Inspection_Start': '',
+    'Inspection_End': '',
+    'Inspection_Closed_Business': 'A',
+    'Violation_Points': '',
+    'Violation_Red_Points': '',
+    'Violation_Descr': '',
+    'Fuzzy_Search': 'N',
+    'Sort': 'H'
+}
+
+
+def get_inspection_page(**kwargs):
+    """ Makes a request to the King County Server, fetches
+    search results, takes query parameters as arguments, returns the
+    content and encoding, if problem raises an error.
+    """
+
+    url = INSPECTION_DOMAIN + INSPECTION_PATH
+    params = INSPECTION_PARAMS.copy()
+    for key, val in kwargs.items():
+        if key in INSPECTION_PARAMS:
+            params[key] = val
+    response = requests.get(url, params=params)
+    response.raise_for_status()
+    return response.content, response.encoding
+
+
+def load_inspection_page(**kwargs):
+    """ The function: reads the inpection_page.html file; returns
+    the content and encoding.
+    """
+
+    files = io.open(paths, 'response_body')
+    response = files.read()
+    files.close()
+    encoding = 'utf-8'
+    return response, encoding
+
+
+def parse_source(html, encoding='utf-8'):
+    """ Sets up HTML as DOM nodes for
+    scraping, takes the response body from the previous function, parses
+    it using BeautifulSoup, returns the parsed object for processing.
+    """
+    parsed = BeautifulSoup(html, 'html5lib', from_encoding=encoding)
+    return parsed
+
+
+def extract_data_listings(html):
+    """ Takes parsed HTML and returns a list of the container nodes."""
+
+    id_finder = re.compile(r'PR[\d]+~')
+    return html.find_all('div', id=id_finder)
+
+
+def has_two_tds(element):
+    """ Takes an element as an argument, returns true if the element is 
+    both a <tr> and contains two <td> elements within it.
+    """
+
+    element_is_tablerow = element.name == 'tr'
+    element_has_tabledata_children = elem.find_all('td', recursive=False)
```

Comment:
NameError here -- you should be catching these either from trying to run the script, or from your linter.